src/mxarray.jl

```diff
@@ -406,13 +406,13 @@ end
 # char arrays and string
 
 function mxarray(s::String)
-    utf16string = transcode(UInt16, s)
+    utf16string = transcode(mxChar, s)
```

Comment:
Ok if you feel that way close this and let's move on. I just want to move on and not create a big issue. I just felt that this was more consistent since it's used  below https://github.com/musm/MATLAB.jl/blob/6993d1e50f38ae73ec45a0bf7a4bbe64ac4058ca/src/mxarray.jl#L646

to convert the String back to Julia.